### PR TITLE
Set up KtLint

### PR DESCRIPTION
**Pull request by shubertm**
The aim of this PR is to enforce Kotlin code style across Memo developers using KtLint plugin.And this makes it easier for new developers too, they can concentrate on solving the actual problem instead of spending much time on formatting code. In most ARK Kotlin repos we are using https://github.com/JLLeitschuh/ktlint-gradle, here as well

### CODE REVIEW

**Comment by tuancoltech** [comment]
@shubertm Could you describe the changes in this PR please?

**Review by hieuwu — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
     <uses-permission android:name="android.permission.RECORD_AUDIO"/>
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
```
Comment (tuancoltech) [inline app/src/main/AndroidManifest.xml:13]: @shubertm For which reason do we need this permission?

File: app/src/main/java/dev/arkbuilders/arkmemo/utils/ContextExt.kt [hunk 2]
```diff
@@ -10,18 +10,20 @@ import androidx.activity.result.ActivityResultLauncher
 fun Context.openLink(url: String) {
     try {
         startActivity(
-            Intent(Intent.ACTION_VIEW).setData(Uri.parse(url)))
+            Intent(Intent.ACTION_VIEW).setData(Uri.parse(url)),
+        )
```
Comment (tuancoltech) [inline app/src/main/java/dev/arkbuilders/arkmemo/utils/ContextExt.kt:14]: @shubertm I don't see the point of adding an extra comma to the end of this line, or in any other parameterized blocks. 
Could you elaborate?

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (shubertm) [reply app/src/main/AndroidManifest.xml:13]: You can check the lint error here https://github.com/ARK-Builders/ARK-Memo/actions/runs/11668539341/job/32488691548. Seems glide posts some notifications. Yet am not really sure

**Pull request merged**